src/utils/withAuth.tsx

```diff
@@ -22,8 +22,12 @@ const withAuth = (WrappedComponent: any, isAdminRightsNeeded?: boolean) => {
     const router = useRouter()
 
     useEffect(() => {
-      if (isAdminRightsNeeded && session && !session.isAdmin) {
-        router.push('/')
+      if (session) {
+        if (session.isApplicant) {
+          router.push('/form/update-info')
```

Comment:
KIV for this first because I need to get Zeyu's work in before I can fully test the user flow.